Doc/whatsnew/3.12.rst

```diff
@@ -236,6 +236,12 @@ New Features
   an additional metaclass argument.
   (Contributed by Wenzel Jakob in :gh:`93012`.)
 
+* Add new function :c:func:`PyFunction_SetVectorcall` to the C API
+  which sets the vectorcall field of a given :c:type:`PyFunctionObject`
+  Warning: extensions using this API must preserve the behavior
+  of the unaltered function!
```

Comment:
I don't think that this document is the right place to put a warning. Why this warning is not in the documentation of the function?